src/js/views/forms/form/form_views.js

```diff
@@ -16,53 +16,60 @@ const ContextTrailView = View.extend({
     <a class="js-back form__context-link">
       {{fas "chevron-left"}}{{ @intl.forms.form.formViews.contextBackBtn }}
     </a>
-    {{fas "chevron-right"}}{{ patient.first_name }} {{ patient.last_name }} - {{ name }}
+    {{fas "chevron-right"}}{{ patient.first_name }} {{ patient.last_name }} - {{ form.name }}
     <div class="form__actions">
+      {{#if showHistory}}<span class="js-history-button form__actions--button{{#if historyResponseId}} is-selected{{/if}}">{{far "history"}}</span>{{/if}}
       <button class="js-expand-button form__actions--button">{{#if isSidebarVisible}}{{far "expand-alt"}}{{else}}{{far "compress-alt"}}{{/if}}</button>
       <button class="js-print-button form__actions--button">{{far "print"}}</button>
       <button class="js-sidebar-button form__actions--button{{#if isActionShown}} is-selected{{/if}}">{{far "file-alt"}}</button>
     </div>
   `,
-  initialize({ patient, state, action }) {
-    this.state = state;
+  initialize({ patient, action, form }) {
     this.patient = patient;
     this.action = action;
-
-    this.listenTo(state, {
-      'change:sidebar': this.render,
-    });
+    this.form = form;
+  },
+  modelEvents: {
+    'change:sidebar': 'render',
+    'change:historyResponseId': 'render',
   },
   ui: {
     sidebarButton: '.js-sidebar-button',
     printButton: '.js-print-button',
     expandButton: '.js-expand-button',
+    historyButton: '.js-history-button',
   },
   triggers: {
     'click .js-back': 'click:back',
     'click @ui.sidebarButton': 'click:sidebarButton',
     'click @ui.printButton': 'click:printButton',
     'click @ui.expandButton': 'click:expandButton',
+    'click @ui.historyButton': 'click:historyButton',
   },
   onClickBack() {
     Radio.request('history', 'go:back');
   },
   templateContext() {
     const patient = this.getOption('patient');
-    const isSidebarVisible = !!this.state.get('sidebar');
+    const isSidebarVisible = !!this.model.get('sidebar');
+    const showHistory = this.model.get('historyResponseId') !== false;
```

Comment:
I think rather that having null and false, this could always be either null or the id.  Then `showHistory` would be based off of `!!action.getFormResponses().length`

Reply:
In this case just relying on `!!action.getFormResponses().length` isn't enough. The history button is also hidden when the form is being updated. In that case we need some indication that we're in that state, hence the `false` value.

Reply:
Why are we hiding it during an update?  It doesn't seem like it would need to be.

Reply:
really though this never actually needs to be the id as it'll only ever be `first()` and then we're not really using it anyways..  it's either 2 states.. show history section or not..  or it's three.. those 2 plus a hidden state..  but I'm kinda hoping hidden is only based on if there's 0 history to show.